keps/sig-api-machinery/2161-apiserver-default-labels/README.md

```diff
@@ -381,7 +381,7 @@ need for new API semantics in the networkpolicy specification, which is already
 
 For example, you may need a way to declare namespaceAsName selectors, which is distinct from namespace label selectors.
 Introducing a new field has its own drawbacks: The obvious API, when combined with a podSelector, decreases security
-of a pod - an old client would be more open than users would expect them to be.
+of a pod - an old client would be more open than users would expect them to be (in the most obvious implementations).  We note that there are workarounds involving special "always off" selectors, which can avoid this scenario, but they come with their own obvious inherent technical debt and costs.
```

Comment:
> own obvious inherent technical debt and costs

this is fairly opinionated.  I don't know that I see significant technical debt here.